mava/wrappers/jumanji.py

```diff
@@ -217,3 +220,128 @@ def observation_spec(self) -> specs.Spec[Union[Observation, ObservationGlobalSta
             )
 
         return spec
+
+
+class MultiCVRPWrapper(Wrapper):
+    """Wrapper for MultiCVRP environment."""
+
+    def __init__(self, env: MultiCVRP, has_global_state: bool = False):
+        self.num_agents = env._num_vehicles
+        self._env = env
+        self.has_global_state = has_global_state
+
+    def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep]:
+        state, timestep = self._env.reset(key)
+        timestep = self.modify_timestep(timestep, state.step_count)
+        return state, timestep
+
+    def step(self, state: State, action: chex.Array) -> Tuple[State, TimeStep]:
+        state, timestep = self._env.step(state, action)
+        timestep = self.modify_timestep(timestep, state.step_count)
+        return state, timestep
+
+    def modify_timestep(self, timestep: TimeStep, step_count: chex.Array) -> TimeStep[Observation]:
+        # avoided the MultiAgentWrapper wrapper to use the step_count provided by the environment
+        observation, global_observation = self._format_observation(timestep.observation)
+        obs_data = {
+            "agents_view": observation,
+            "action_mask": timestep.observation.action_mask,
+            "step_count": jnp.repeat(step_count, (self.num_agents)),
+        }
+        if self.has_global_state:
+            obs_data["global_state"] = global_observation
+            observation = ObservationGlobalState(**obs_data)
+        else:
+            observation = Observation(**obs_data)
+
+        reward = jnp.repeat(timestep.reward, (self.num_agents))
+        discount = jnp.repeat(timestep.discount, (self.num_agents))
+        timestep = timestep.replace(observation=observation, reward=reward, discount=discount)
+        return timestep
+
+    def _format_observation(
+        self, observation: MultiCVRPState
+    ) -> Tuple[chex.Array, Union[None, chex.Array]]:
+        """
+        Formats the observation dictionary from the environment into a format suitable for mava.
+
+        Args:
+            observation (Dict[chex.Array]): The raw observation dict provided by jumanji
```

Comment:
Observation is a named tuple
```suggestion
        Concatenates all observation fields into a single array.

        Args:
            observation (MultiCvrpObservation): The raw observation NamedTuple provided by jumanji.
```